### ISSUE TITLE
generate dynamic gradle/maven/... dependency and repository component

### ISSUE THREAD

**Issue by atsu85** [jitpack/jitpack.io#380]
It would be nice, if github project could add smth simple to the README.md to generate dynamic gradle/maven/sbt... dependency and repository component to the markdown page, just like the example in https://jitpack.io/# (under "How to" that allows selecting gradle/maven/sbt/lein dependency manager and shows code for adding repository + dependency)?

Unfortunately github removes script tags, so injecting JS to generate the component doesn't seem to be an option.

Another idea worth investigating would be using iframe - group, artifact and version could be passed as url parameters to get back dynamic html content with correct artifact information.

**Comment by atsu85** [comment]
To clear things a bit more - i was hoping that jitpack could host a page smth like
https://jitpack.io/dependencySampleWidget?group=com.github.User&artifact=Repo%20name&version=Release%20tag
that generates html page that contains component just like the example in https://jitpack.io/# (under "How to" that allows selecting gradle/maven/sbt/lein dependency manager and shows code for adding repository + dependency).

If you'd tweak the url parameters, you should get smth different.

**Comment by jitpack-io** [comment]
Hi

That is a brilliant idea and a common use case. Lots of projects have websites and show the same dependency information.

Starting to test embeddable code snippets:
`https://www.jitpack.io/v/User/Project.html`

For example https://www.jitpack.io/v/jitpack/gradle-simple.html
```html
<iframe src="https://www.jitpack.io/v/jitpack/gradle-simple.html"
width="100%" height="400" frameBorder="0">
</iframe>
```

**Comment by atsu85** [comment]
Thanks, what about multi-module projects? Is this 
https://www.jitpack.io/v/raphaeljolivet.java2typescript/java2typescript-maven-plugin.html
supposed to work for
https://github.com/raphaeljolivet/java2typescript#installation
?

**Comment by jitpack-io** [comment]
Yes, planning to support modules as well. Will probably show a module selector and have an option to show a default module.
Wip: https://www.jitpack.io/v/jitpack/gradle-modular.html

**Comment by atsu85** [comment]
Thanks, looks promising. I have 2 questions tough:
1) how can i select specific subproject by url
2) What is the reason for not using group and artifact id's in url? Maybe I'm missing smth important, but it seems to me, that this way generating the component would be easier - It seems to me, that from the group and artifact you can find out even more information, including the Github repository, its subprojects and latest released version, that You are already using for that component.

**Comment by jitpack-io** [comment]
For a subproject you can use the full groupId:
https://www.jitpack.io/v/com.github.jitpack.gradle-modular/client.html

It also works for single module projects:
https://www.jitpack.io/v/com.github.jitpack/gradle-simple.html
but since 'com.github' is the default group prefix, it can be omitted for simple projects.

**Comment by atsu85** [comment]
Thanks, works as expected :)
Should i avoid using it just for now (is this likely to break in the near future) or do You plan to use another url if there are breaking changes?

I'd like to use it for https://www.jitpack.io/v/com.github.raphaeljolivet.java2typescript/java2typescript-maven-plugin.html but as i'm not going to visit that page often, i might miss the breaking changes (that would eventually show up in the github page).

**Comment by jitpack-io** [comment]
It's currently in beta mode and being tests so might change but it's unlikely.
If it does then will post a comment here.
